hoot-core/src/main/cpp/hoot/core/visitors/AddElementIdVisitor.cpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * This file is part of Hootenanny.
+ *
+ * Hootenanny is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * --------------------------------------------------------------------
+ *
+ * The following copyright notices are generated automatically. If you
+ * have a new notice to add, please use the format:
+ * " * @copyright Copyright ..."
+ * This will properly maintain the copyright information. Maxar
+ * copyrights will be updated automatically.
+ *
+ * @copyright Copyright (C) 2022 Maxar (http://www.maxar.com/)
+ */
+#include "AddElementIdVisitor.h"
+
+// hoot
+#include <hoot/core/schema/MetadataTags.h>
+#include <hoot/core/util/Factory.h>
+
+namespace hoot
+{
+
+HOOT_FACTORY_REGISTER(ElementVisitor, AddElementIdVisitor)
+
+void AddElementIdVisitor::visit(const ElementPtr& pElement)
+{
+  Tags& tags = pElement->getTags();
+  tags[MetadataTags::HootId()] = QString::number(pElement->getId());
```

Comment:
I think we need an indicator of the table (Node, Way, Relation) as well as the numeric ID.
`getElementId().toString()` provides this `Way(758208161)`

This unit test should also assert the format of the `hoot:id`.

Reply:
Format has been changed. Looking at how to test the structure.

Reply:
After a number of changes, the test file now looks for a single verbose value as well as the "hoot:id" tag.